frontend/dashboard/src/RelayEnvironment.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Environment,
+  Network,
+  RecordSource,
+  Store,
+  FetchFunction,
+} from "relay-runtime";
+import keycloak from "./keycloak";
+
+const HTTP_ENDPOINT = "https://workflows.diamond.ac.uk/graphql";
+
+const kcinit = keycloak.init({
+  onLoad: "login-required"
+})
+.then(
+  auth => {
+      console.info("Authenticated");
+      console.log("auth", auth);
+      keycloak.onTokenExpired = () => {
+        console.log("token expired");
+      };
+  },
+  () => {
+    console.error("Authentication failed");
+  }
+);
+
+const fetchFn: FetchFunction = async (request, variables) => {
+  if (!keycloak.authenticated) {
+    await kcinit;
+  }
+
+  if (keycloak.token) {
+    const resp = await fetch(HTTP_ENDPOINT, {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer ${keycloak.token}`,
+        Accept:
+          "application/graphql-response+json; charset=utf-8, application/json; charset=utf-8",
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        query: request.text, // <-- The GraphQL document composed by Relay
+        variables,
+      }),
+    });
+  
+    // This is following the official relay docs: 
+    // https://relay.dev/docs/getting-started/step-by-step-guide/#22-a-fetchgraphql-helper
+    // Couldn't find a way that avoids returning type Promise<any>
+    return await resp.json(); // eslint-disable-line @typescript-eslint/no-unsafe-return
```

Comment:
This is fine, relay expects a `Promise<any>` and does the typechecking itself at runtime afaik

Reply:
I tried `return await (resp.json() as Promise<any>); ` but the eslint was not happy

```
  51:19  error  This assertion is unnecessary since it does not change the type of the expression  @typescript-eslint/no-unnecessary-type-assertion
  51:42  error  Unexpected any. Specify a different type                                           @typescript-eslint/no-explicit-any
```